src/server/server_family.cc

```diff
@@ -1577,11 +1582,10 @@ void ServerFamily::Info(CmdArgList args, ConnectionContext* cntx) {
     } else {
       append("role", "replica");
 
-      // it's safe to access replica_ because replica_ is created before etl.is_master set to
-      // false and cleared after etl.is_master is set to true. And since the code here that checks
-      // for is_master and copies shared_ptr is atomic, it1 should be correct.
-      auto replica_ptr = replica_;
-      Replica::Info rinfo = replica_ptr->GetInfo();
+      // The replica pointer can still be mutated even while master=true,
+      // we don't want to drop the replica object in this fiber
+      unique_lock lk{replicaof_mu_};
+      Replica::Info rinfo = replica_->GetInfo();
```

Comment:
I did not investigate 😢 But when I dropped it here all kinds of fiber / eventcount asserts failed badly

Reply:
I believe this is because of ABSL_GUARDED_BY
std::shared_ptr<Replica> replica_ ABSL_GUARDED_BY(replicaof_mu_);
@chakaz any idea why Vlad sees now this asserts fail but we didnt see it before?

Reply:
I'm slightly out of context, but when you say asserts you mean at runtime, right?
`ABSL_GUARDED_BY` is a compile-time-only feature, so it shouldn't affect runtime